### PR TITLE
Disable updating system headers while snapshotting - Closes #2261

### LINKED ISSUE #2261: RPC status call gives wrong height when syncing
### Expected behavior

The RPC status call returns the same height as the HTTP during sync

### Actual behavior

The RPC status call returns a fixed height and broadhash for a very ling time, even if the node's height increases during sync.

In this example, I called the RPC status every 2 seconds on my node between 20:12:01 and 20:36:40 UTC. The returned status increases only a handful of times and I get the same height for a very long period of time

| height | number of status responses |
| ----| ----|
|  5030757 | 57 |
| 5038050 | 149 |
| 5039766 | 51 |
| 5041515 | 405 |
| 5044650 | 78 |

* [Series of status responses](https://gist.github.com/webmaster128/bbf730f669fea48d4411d612c494d290)
* [Logs of the syncing node](https://gist.github.com/webmaster128/9e5aa466be0cbbe6459f952355c98bdf)

### Steps to reproduce

Repeatedly run the RPC status call on a given node that is syncing

### Which version(s) does this affect? (Environment, OS, etc...)

Testnet, 1.0.0-rc.1

### PR DESCRIPTION
**Pull request by yatki**
### What was the problem?
As @4miners stated here https://github.com/LiskHQ/lisk/pull/2300#issuecomment-412379504: 

> After the change we're always calling modules.system.update from inside Verify.prototype.processBlock. System.prototype.update is calling System.prototype.getBroadhash which is performing database query SELECT * FROM blocks_list ORDER BY b_height DESC LIMIT 5 OFFSET 0 and calculate SHA-256. While both the query (0.5ms) and hash operation are fast - they are executed for every block. Those operations are not needed for snapshotting and will decrease the performance:
6000000 blocks * 0.5 ms = 3000000 ms = 3000 s = 50 minutes.
So change introduced in this PR will make snapshotting process longer by at least 50 minutes (only because of the query). Please consider not executing those when snapshotting.

### How did I fix it?
I disabled updating system headers while snapshotting.

### Review checklist

* The PR resolves #2261 
* All new code is covered with unit tests
* All new code was formatted with Prettier
* Linting passes
* Tests pass
* Commit messages follow the [commit guidelines](CONTRIBUTING.md#git-commit-messages)
* Documentation has been added/updated

### CODE REVIEW

File: modules/blocks/verify.js [hunk 1]
```diff
@@ -45,14 +45,15 @@ __private.lastNBlockIds = [];
...
-	constructor(logger, block, transaction, db) {
+	constructor(logger, block, transaction, db, config) {
 		library = {
 			logger,
 			db,
 			logic: {
 				block,
 				transaction,
 			},
+			config,
```
Comment (4miners) [inline modules/blocks/verify.js]:
```
config: {
	loading: {
		snapshotRound: config.loading.snapshotRound,
	},
},
```

File: modules/blocks/verify.js [hunk 2]
```diff
@@ -821,7 +822,9 @@ Verify.prototype.processBlock = function(block, broadcast, saveBlock, cb) {
 			// 'false' if block comes from chain synchronisation process
 			updateSystemHeaders(seriesCb) {
 				// Update our own headers: broadhash and height
-				modules.system.update(seriesCb);
+				!library.config.loading.snapshotRound
```
Comment (4miners) [inline modules/blocks/verify.js]: Maybe the opposite?
`library.config.loading.snapshotRound ? seriesCb() : modules.system.update(seriesCb);`

File: test/unit/modules/blocks/verify.js [hunk 3]
```diff
@@ -1931,6 +1937,44 @@ describe('blocks/verify', () => {
 			done();
 		});
 
+		describe('system headers', () => {
```
Comment (4miners) [inline test/unit/modules/blocks/verify.js]: This file is for unit tests of `modules/blocks/verify.js`, you shouldn't add describe block for `system headers`, but add corresponding paths in `processBlock` tests.

File: test/unit/modules/blocks/verify.js [hunk 4]
```diff
@@ -1942,7 +1986,6 @@ describe('blocks/verify', () => {
 						saveBlock,
 						err => {
 							expect(err).to.be.null;
-							expect(modules.system.update.calledOnce).to.be.true;
```
Comment (4miners) [inline test/unit/modules/blocks/verify.js]: We should still test those.

File: modules/blocks/verify.js (hunk 2 above)
Comment (yatki) [reply modules/blocks/verify.js]: I considered that too. But I'd like to provide the desired functionality in the first part and I'd prefer to keep the same structure with rest of the short-hand if usages.

File: test/unit/modules/blocks/verify.js (hunk 3 above)
Comment (yatki) [reply test/unit/modules/blocks/verify.js]: It's consistent with the current suite structure.

File: modules/blocks/verify.js (hunk 1 above)
Comment (yatki) [reply modules/blocks/verify.js]: @4miners with this usage we will override the config object and loose the other properties?

Comment (4miners) [reply modules/blocks/verify.js]: Yep

File: test/unit/modules/blocks/verify.js (hunk 4 above)
Comment (yatki) [reply test/unit/modules/blocks/verify.js]: We already cover that case here https://github.com/LiskHQ/lisk/blob/0206f72e80fad817b48e7fe8dc420d54b7af72cb/test/unit/modules/blocks/verify.js#L1941

**Pull request merged**